components/datetime/tests/datetime.rs

```diff
@@ -645,68 +606,26 @@ fn test_time_zone_patterns() {
         } in &test.expectations
         {
             for pattern_input in patterns {
-                let new_pattern1: runtime::Pattern = pattern_input.parse().unwrap();
-                let new_pattern2: runtime::Pattern = pattern_input.parse().unwrap();
-                time_patterns_data.payload.with_mut(move |data| {
-                    data.time_h11_h12.medium = new_pattern1;
-                    data.time_h23_h24.medium = new_pattern2;
-                });
-                let local_provider = MultiForkByMarkerProvider::new(vec![
-                    AnyPayloadProvider::from_payload::<GregorianDateSymbolsV1Marker>(
-                        symbols_data.payload.clone(), //
-                    ),
-                    #[cfg(feature = "experimental")]
-                    AnyPayloadProvider::from_payload::<DateSkeletonPatternsV1Marker>(
-                        skeleton_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<GregorianDateLengthsV1Marker>(
-                        date_patterns_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<TimeLengthsV1Marker>(
-                        time_patterns_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<WeekDataV1Marker>(
-                        week_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<DecimalSymbolsV1Marker>(
-                        decimal_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<TimeZoneFormatsV1Marker>(
-                        time_zone_formats_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<MetazoneSpecificNamesShortV1Marker>(
-                        metazone_specific_short_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<MetazoneSpecificNamesLongV1Marker>(
-                        metazone_specific_long_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<MetazoneGenericNamesShortV1Marker>(
-                        metazone_generic_short_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<MetazoneGenericNamesLongV1Marker>(
-                        metazone_generic_long_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<ExemplarCitiesV1Marker>(
-                        exemplar_cities_data.payload.clone(), //
-                    ),
-                ]);
-
+                let parsed_pattern = DateTimePattern::try_from_pattern_str(pattern_input).unwrap();
```

Comment:
I feel like I deserve a spot bonus for finally getting rid of this terrible pattern override code that's been haunting us for so long